components/postcode-lookup/index.js

```diff
@@ -0,0 +1,339 @@
+        enterManually: conditionalTranslate('pages.address.lookup.enter-manually', req.translate) ||
+          defaults.ENTER_MANUALLY,
+        postcodeHeading: conditionalTranslate('pages.postcode.postcode-heading', req.translate) ||
+          defaults.POSTCODE_HEADING,
+        addressLookupHeading: conditionalTranslate('pages.address-lookup.address-lookup-heading', req.translate) ||
+          defaults.ADDRESS_LOOKUP_HEADING,
+        manualAddressHeading: conditionalTranslate('pages.address-lookup.manual-address-heading', req.translate) ||
+          defaults.MANUAL_ADDRESS_HEADING,
+        manualAddressParagraph: conditionalTranslate('pages.address-lookup.manual-address-paragraph', req.translate) ||
+          defaults.MANUAL_ADDRESS_PARAGRAPH,
+        searchErrorHeading: conditionalTranslate('pages.postcode-search-problem.searchErrorHeading', req.translate) ||
+          defaults.SEARCH_ERROR_HEADING,
+        postcodeEntered: sessionPostcode,
+        addressesExist: (req.sessionModel.get('addresses') !== undefined &&
+        req.sessionModel.get('addresses').length > 0) ? true : false,
+        noAddressHeading: conditionalTranslate('pages.postcode-search-problem.noAddressHeading', req.translate) ||
+          defaults.NO_ADDRESS_HEADING,
+        addressCount: addressCount,
+        postcodeSearchTitle: conditionalTranslate('pages.postcode-search.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_SEARCH_TITLE,
+        addressLookupTitle: conditionalTranslate('pages.address-lookup.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_TITLE,
+        addressDetailsTitle: conditionalTranslate('pages.address-details.title', req.translate) ||
+          defaults.ADDRESS_DETAILS_TITLE,
+        lookupProblemTitle: conditionalTranslate('pages.address-lookup-problem.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_PROBLEM_TITLE,
+        route: this.options.route,
+        editLink,
+        cantFind,
+        searchByPostcodeLink,
+        postcodeError,
+        section
+      });
+    }
+
+    postcode(req, res, callback) {
+      // Clear the value stored in the addresses radio button group
+      req.sessionModel.set(`${addressKey}-select`, '');
+      // Call OS Places API to return list of addresses by postcode
+      const enteredPostcode = req.form.values[`${addressKey}-postcode`];
+
+      axios.get(apiURL + '?postcode=' +  enteredPostcode + '&key=' + apiKey)
+        .then(function (response) {
+          const addressList = [];
+          _.forOwn(response.data.results, function (value) {
+            addressList.push(value.DPA);
+          });
+
+          req.sessionModel.set(`${addressKey}-addresses`, addressList);
+          return callback();
+        }).catch(error => {
+          if(error.response.status === '404' || error.response.status === '429' ||
+            error.response.status === '503' || error.response.status === '500') {
+            req.query.step === 'postcodeSearchIssue';
```

Comment:
too generic, are there detailed info on the API documentation on what the errors means?

Reply:
Added logger within the block of code where the function is called and got rid of the 'if' block from the code above to remove duplication. Also, modified the logic so that instead of checking for these specific error codes, it will now log the error regardless of the error code.